File: backend/seguridad.py
from datetime import datetime, timedelta
from jose import jwt, JWTError
from passlib.context import CryptContext
from configuracion import JWT_SECRET, JWT_EXPIRA_MINUTOS

ALGORITMO = "HS256"
contexto_cripto = CryptContext(schemes=["bcrypt"], deprecated="auto")


def hashear_password(password: str) -> str:
    return contexto_cripto.hash(password)


def verificar_password(password: str, hash_guardado: str) -> bool:
    return contexto_cripto.verify(password, hash_guardado)


def crear_token(datos: dict) -> str:
    payload = datos.copy()
    expira = datetime.utcnow() + timedelta(minutes=JWT_EXPIRA_MINUTOS)
    payload.update({"exp": expira})
    return jwt.encode(payload, JWT_SECRET, algorithm=ALGORITMO)


def decodificar_token(token: str) -> dict:
    try:
        return jwt.decode(token, JWT_SECRET, algorithms=[ALGORITMO])
    except JWTError:
        return {}

File: backend-node/src/seguridad.js
import bcrypt from 'bcryptjs';
import jwt from 'jsonwebtoken';

const JWT_SECRET = process.env.JWT_SECRET || 'secreto_desarrollo';
const JWT_EXPIRA_MINUTOS = parseInt(process.env.JWT_EXPIRA_MINUTOS || '480');

export function hashearPassword(password) {
  return bcrypt.hashSync(password, 10);
}

export function verificarPassword(password, hash) {
  return bcrypt.compareSync(password, hash);
}

export function crearToken(datos) {
  return jwt.sign(datos, JWT_SECRET, { expiresIn: JWT_EXPIRA_MINUTOS * 60 });
}